validphys2/src/validphys/dataplots.py

```diff
@@ -921,6 +923,106 @@ def plot_replica_sum_rules(pdf, sum_rules, Q):
     return fig
 
 
+@figure
+def plot_polarized_momentum(pdf, Q, xmin=0.001, angular_momentum=False):
+    """
+    Plot the correlated uncertainties for the truncated integrals of the polarized
+    gluon and singlet distributions.
+    """
+    from matplotlib.patches import Ellipse
+
+    def compute_hists(preds, nbins=10):
+        binning = np.linspace(preds.min(), preds.max(), num=nbins)
+        frequencies, bins = np.histogram(preds, bins=binning, density=True)
+        return {"x": bins[:-1], "bins": bins, "weights": frequencies}
+
+    def compute_ellipse(preds_x, preds_y, nstd=3):
+        covmat = np.cov(preds_x, preds_y)
+        eigval, eigvec = np.linalg.eig(covmat)
+        sqrt_eigval = np.sqrt(eigval)
+
+        return {
+            "xy": (preds_x.mean(), preds_y.mean()),
+            "width": 2.0 * nstd * sqrt_eigval[0],
+            "height": 2.0 * nstd * sqrt_eigval[1],
+            "angle": np.degrees(np.arctan2(*eigvec[:, 0][::-1])),
+            "fill": False,
+            "linewidth": 2.0,
+            "color": "C0",
+        }
+
+    predictions = polarized_sum_rules(pdf, Q, lims=[(xmin, 1)])
```

Comment:
Mabye there is something I'm missing here, but shouldn't it be better if `polarized_sum_rules` where to return both the very low x and the normal values all the time? The computation is cheap enough that I think it would be worthwile, that way it can be lifted from here to the arguments of the function and leave reportengine deal with it.

(take it as a... separate suggestion with respect to the others because I'm not sure it would deal with it _correctly_ and this plotting function is self-contained enough that I'd be willing to accept these 2 calls to `polarized_sum_rules`)

Reply:
Sure.
As far as I can see you have two calls to `polarized_sum_rules(pdf, Q, lims=[(xmin, 1)])` and `polarized_sum_rules(pdf, Q, lims=[(1e-4, xmin)])`. 
I am wondering whether it makes sense to make `polarized_sum_rules` to always compute the sum rules for `(xmin, 1)` and `(xmin*1e-1, xmin)`. (maybe it doesn't if this is used in other places).

As a separate note, I think it would be better to have `polarized_sum_rules` as an argument to the function, so you will get already the value of `polarized_sum_rules(pdf, Q, lims=[(xmin, 1)])` as computed by validphys.
Then if you need to compute also `polarized_sum_rules(pdf, Q, lims=[(1e-4, xmin)])` that is a separate call.

But as I said, I'm willing to accept it as it is if that would complicate things, the only problematic thing is the issue I've mentioned about `xmin`.

Reply:
I think there is a nicer compromise in [6edffdb](https://github.com/NNPDF/nnpdf/pull/1979/commits/6edffdbc201843a947a5f89ee854e07c03bb8599), let me know if you like this better xd

Reply:
If this works (namely, doesn't break things elsewhere) this is ok for me.

(if it does, a good middle way is to perform the call explicitly in the functions that call the plotting)